File: app/src/main/java/com/hardening/app/model/ApiResponse.java
package com.hardening.app.model;

import com.google.gson.annotations.SerializedName;

/**
 * 后端统一响应包装。
 * JSON 形如：{"code": 0, "message": "ok", "data": {...}}
 * code == 0 表示成功，其余值为业务错误码。
 *
 * @param <T> data 字段的实际类型
 */
public class ApiResponse<T> {

    @SerializedName("code")
    private int code;

    @SerializedName("message")
    private String message;

    @SerializedName("data")
    private T data;

    public int getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }

    public T getData() {
        return data;
    }

    public boolean isSuccess() {
        return code == 0;
    }
}

File: backend/app/common/traits/ApiResponse.php
<?php
declare(strict_types=1);

namespace app\common\traits;

use think\Response;

/**
 * API 统一响应 Trait
 */
trait ApiResponse
{
    /**
     * 成功响应
     */
    protected function success($data = null, string $message = 'success', int $code = 0): Response
    {
        return $this->json($code, $message, $data);
    }

    /**
     * 失败响应
     */
    protected function fail(string $message = 'failed', int $code = 1, $data = null, int $httpStatus = 200): Response
    {
        return $this->json($code, $message, $data, $httpStatus);
    }

    /**
     * 统一JSON输出 {code, message, data}
     */
    protected function json(int $code, string $message, $data = null, int $httpStatus = 200): Response
    {
        $payload = [
            'code'    => $code,
            'message' => $message,
            'data'    => $data,
        ];
        return Response::create($payload, 'json', $httpStatus);
    }

    /**
     * 从 JWT 中间件注入的请求上下文获取已认证用户 ID。
     *
     * 严格以来源 JWT 的 userId 为准，禁止通过 X-User-Id 等请求头覆盖，
     * 否则持有任意有效 token 的攻击者可凭 X-User-Id 任意冒用其他用户身份。
     */
    protected function userId(): int
    {
        $uid = request()->userId ?? 0;
        return (int) $uid;
    }

    /**
     * 分页参数
     */
    protected function pageParams(): array
    {
        $page = (int) request()->param('page', 1);
        $size = (int) request()->param('page_size', 15);
        $page = max(1, $page);
        $size = min(100, max(1, $size));
        return [$page, $size];
    }
}
